application_store_api/namespace_fund/data_store/store.py

```diff
@@ -0,0 +1,117 @@
+
+        if application_fund_name not in self.funds:
+            # create a new fund entry
+            self.funds[application_fund_name] = []
+
+        # place application within the fund
+        self.funds[application_fund_name].append(application)
+        return application
+
+    def get_applications_for_fund(
+        self, fund_name, datetime_start, datetime_end
+    ):
+        try:
+            fund_data = self.funds[fund_name]
+            applications_within_period = []
+            if datetime_start and datetime_end:
+                # convert string dates into datetimes
+                start = date_parser.parse(datetime_start).astimezone(UTC)
+                end = date_parser.parse(datetime_end).astimezone(UTC)
+
+                # compare period limits against application dates within fund
+                for application in self.funds[fund_name]:
+                    if (
+                        start
+                        < application["date_submitted"].astimezone(UTC)
+                        < end
+                    ):
+                        applications_within_period.append(application)
+                return json.loads(
+                    json.dumps(applications_within_period, default=str)
+                )
+            else:
+                return json.loads(json.dumps(fund_data, default=str))
+        except Exception:
+            return f"Fund: {fund_name} not found.", 400
+
+    def get_application_by_id(self, fund_name, application_id):
+
+        try:
+            for application in self.funds[fund_name]:
+                if application["id"] == application_id:
+                    return json.loads(json.dumps(application, default=str))
+            return (
+                f"Application id: {application_id} not found in fund:"
+                f" {fund_name}",
+                400,
+            )
+        except Exception:
+            return f"Fund: {fund_name} not found.", 400
+
+    def delete_application_by_id(self, fund_name, application_id):
+
+        try:
+            for application in self.funds[fund_name]:
+                if application["id"] == application_id:
+                    self.funds[fund_name].remove(application)
+                    return f"{application_id} deleted", 204
+            return (
+                f"Application id: {application_id} not found in fund:"
+                f" {fund_name}",
+                400,
+            )
+        except Exception:
+            return f"Fund: {fund_name} not found.", 400
+
+    def delete_all(self, delete_key):
+
+        if delete_key == "positive-clear":
+            self.funds = {}
+        else:
+            return "No key provided. Clear unsuccessful"
+
+
+# Sample data
+sample_application_data = [funding_service_design_application]
+
+# In memory data object instance
+APPLICATIONS = application_data_access_object()
+
+for application in sample_application_data:
+    APPLICATIONS.create_application(application)
```

Comment:
Could creating initial data be a method of the ApplicationDAO class rather than on its own here (as it's part of the same concept)?

Reply:
good shout! create_application is a method used by the routes. There is actually no use to the block anymore so i have removed. The initial fund data needs testable data (not a changing uuidv4 everytime we restart) so we can test again known data - this is why the create_application method id not called for that data.